### DISCUSSION TITLE
Handling conflicts when applying multiple templates

### DISCUSSION THREAD

**Discussion by lllama** [copier-org/copier#1197] [Q&A]
I am experimenting with applying multiple templates to my project. However, I sometimes have multiple templates containing the same file. It looks like I can't generate `rej` files when using `run_copy` as these are only generated for updates. Is there a way to achieve that I'm looking to do?

**Comment by lllama** [comment]
Having read the docs again, my understanding is that I will have to apply all the templates using `copy` initially and then updates are supported using the appropriate answer files.

Ideally, I'd like to have the update behaviour for multiple templates and let users end up with conflict markers or `.rej` files. Failing that, is there a way to test where a copy would overwrite a file? I could then do some tests to make sure things are committed etc.

**Reply by yajo** [reply]
Conflicts (either in rej or inline mode) only happen on updates, yes.

When (re)copying, Copier just overwrites any file you tell it to.

In general terms, if you use 2 templates, it's better if each of them updates a different set of files.

**Reply by htbecker** [reply]
I am facing the same issues currently and I will not be able to fully separate files.
I would love to have the copy process to offer three options for the overwrite behaviour

- false -> do not overwrite
- true -> overwrite files
- rej -> create the generated files next to the existing one (like conflict=rej)

Would anything speak against such a behaviour? Besides the changing the now boolean flag to a tri-state / string / choices option which would be a breaking change for the cli.

**Reply by yajo** [reply]
See https://github.com/copier-org/copier/pull/1155. Overwriting is no longer optional because it was buggy.

In case you manage to solve the bugs it had, this would be indeed accepted! But let me warn you that it won't be easy.
